### ISSUE TITLE
vmbackuper doesnt work with IAM Role assumed for service account

### ISSUE THREAD

**Issue by k1rk** [VictoriaMetrics/VictoriaMetrics#726]
**Describe the bug**
IAM Role assumption for service accounts (IRSA) is used in AWS EKS. More details [here](https://aws.amazon.com/blogs/opensource/introducing-fine-grained-iam-roles-service-accounts/)

With such approach pod gets two env variables
```
AWS_ROLE_ARN='arn:aws:iam::xxxxxxxxx:role/role-name'
AWS_WEB_IDENTITY_TOKEN_FILE='/var/run/secrets/eks.amazonaws.com/serviceaccount/token'
```
and token mounted to `/var/run/secrets/eks.amazonaws.com/serviceaccount/token`

with this token you expected to call sts and get access key, secret key and session token
```
$ aws sts assume-role-with-web-identity \
 --role-arn $AWS_ROLE_ARN \
 --role-session-name mh9test \
 --web-identity-token file://$AWS_WEB_IDENTITY_TOKEN_FILE \
 --duration-seconds 1000 > /tmp/irp-cred.txt
$ export AWS_ACCESS_KEY_ID="$(cat /tmp/irp-cred.txt | jq -r ".Credentials.AccessKeyId")"
$ export AWS_SECRET_ACCESS_KEY="$(cat /tmp/irp-cred.txt | jq -r ".Credentials.SecretAccessKey")"
$ export AWS_SESSION_TOKEN="$(cat /tmp/irp-cred.txt | jq -r ".Credentials.SessionToken")"
```

AWS SDK doing it under the hood starting from 1.23.13, but as you don't use sdk, this flow should be implemented

Right now if i use service account with backuper pod i just receive `AccessDenied: Access Denied`
while from same pod i'm able to work with s3 via installed aws-cli

**To Reproduce**
start vmbackuper pod with service account that can assume iam role

**Expected behavior**
vmbackuper should be able to assume iam role, and use aws services with it.

**Version**

```
❯ ./vmbackuper -version
Copyright 2020 Victoria Metrics, Inc - All Rights Reserved.
Unauthorized using, copying of this software, via any medium is strictly prohibited.
Proprietary. For any questions please contact info@victoriametrics.com
vmbackuper-20200819-073004-heads-fix-auth-acc-link-0-g5c0f1c666
```

**Used command-line flags**
```
flag{name="allowMissingConfig", value="false"} 1
flag{name="allowUnknownFlags", value="false"} 1
flag{name="concurrency", value="10"} 1
flag{name="config", value=""} 1
flag{name="configFilePath", value=""} 1
flag{name="configProfile", value="default"} 1
flag{name="configUpdateInterval", value="0s"} 1
flag{name="credsFilePath", value=""} 1
flag{name="customS3Endpoint", value=""} 1
flag{name="disableDaily", value="false"} 1
flag{name="disableHourly", value="false"} 1
flag{name="disableMonthly", value="false"} 1
flag{name="disableWeekly", value="false"} 1
flag{name="dst", value="s3://bucket-test/backup/irsa"} 1
flag{name="dumpflags", value="false"} 1
flag{name="enableTCP6", value="false"} 1
flag{name="fs.disableMmap", value="false"} 1
flag{name="http.disableResponseCompression", value="false"} 1
flag{name="http.maxGracefulShutdownDuration", value="7s"} 1
flag{name="http.pathPrefix", value=""} 1
flag{name="http.shutdownDelay", value="0s"} 1
flag{name="httpListenAddr", value=":8300"} 1
flag{name="httpServerMetricsAuthKey", value="secret"} 1
flag{name="httpServerPprofAuthKey", value="secret"} 1
flag{name="httpServerRedirectToHTTPS", value="false"} 1
flag{name="loggerErrorsPerSecondLimit", value="10"} 1
flag{name="loggerLevel", value="INFO"} 1
flag{name="loggerOutput", value="stderr"} 1
flag{name="loggerStackDriverProjectID", value=""} 1
flag{name="maxBytesPerSecond", value="0"} 1
flag{name="memory.allowedBytes", value="0"} 1
flag{name="memory.allowedPercent", value="60"} 1
flag{name="runOnStart", value="false"} 1
flag{name="snapshot.createURL", value="http://localhost:8429/snapshot/create"} 1
flag{name="snapshot.deleteURL", value="http://localhost:8429/snapshot/delete"} 1
flag{name="storageDataPath", value="/victoria-metrics-data"} 1
flag{name="version", value="false"} 1
```

**Comment by valyala** [comment]
@k1rk , thanks for the detailed issue report!

`vmbackuper` reuses the code from `vmbackup`. This code relies on AWS SDK - see https://github.com/VictoriaMetrics/VictoriaMetrics/blob/a034f02fb2e08a5f1d4c3d6cf92d3b9028817734/lib/backup/s3remote/s3.go#L58-L89 for details. I suppose this code has some issues, which prevent from using IAM Role assumed for service account. It would be great if somebody more familiar with assumed IAM Role usage in AWS SDK could help with this issue.

**Comment by k1rk** [comment]
well yes, flow used in this case is a bit more complicated as instead of regular assume role, `sts:assume-role-with-web-identity` is called.
But it was implemented in aws-sdk-go starting from `v1.23.13`
and according to docs this should work in your setup. 
The only possible problem i can see is that you specify `Profile` always, rather than only when profile was specified in args.

**Comment by f41gh7** [comment]
> well yes, flow used in this case is a bit more complicated as instead of regular assume role, `sts:assume-role-with-web-identity` is called.
> But it was implemented in aws-sdk-go starting from `v1.23.13`
> and according to docs this should work in your setup.
> The only possible problem i can see is that you specify `Profile` always, rather than only when profile was specified in args.

yep, seems like profile is a source of problem, can you try to run it with flag `--configProfile=""` ?

**Comment by valyala** [comment]
`vmbackup` and `vmbackuper` use `default` profile if `-configProfile` isn't set. Probably, it should be set to empty string by default if this doesn't break existing configs?

**Comment by valyala** [comment]
@k1rk , the issue must be fixed in the commit https://github.com/VictoriaMetrics/VictoriaMetrics/commit/8d71a60a76f9594ace68ec6d3fa74508797bdfa1 , which will be included in the next release.

**Comment by valyala** [comment]
The bugfix has been included in [v1.41.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.

@k1rk , feel free re-opening the issue if `vmbackup`/`vmrestore`/`vmbackuper` don't work as expected in [v1.41.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases).
